### ISSUE TITLE
1. A referrer should visit /refer-a-mate and receive a unique, shareable URL.

### ISSUE THREAD

**Issue by christoshrousis** [decryption256/sizzle-refer-a-mate#1]
- [ ] On page load, should detect master cookie, or create master cookie.
- [ ] Master cookie should contain: 
- [ ] User Nickname.
- [ ] Email Address.
- [ ] ???Existing Unique URL???
- [ ] If cookie does not exist, display a form that collects data from the user.

Should probably treat the nickname + email as a unique key that is stored server side, and linked to the unique url.

_1. Existing subscriber (or anyone really) visits https://thesizzle.com.au/refer-a-mate and receives a unique URL (e.g: https://thesizzle.com.au?ref=8790z89hb) that they can share and tell their friends to sign up for a free trial of The Sizzle. Also need to collect their nickname (for the leaderboard) and their email address (to contact the winner of the comp)._

**Comment by decryption256** [comment]
Yep, that sounds right. Bonus points if the URL can be prettified, or adjustable by the user so it's easier for them to remember so they spread it wider (and possibly easier for offline marketing), but not vital.
